hil/ext/switches/dellnos9.py

```diff
@@ -0,0 +1,380 @@
+    interface_type = db.Column(db.String, nullable=False)
+
+    @staticmethod
+    def validate(kwargs):
+        schema.Schema({
+            'hostname': basestring,
+            'username': basestring,
+            'password': basestring,
+            'interface_type': basestring,
+        }).validate(kwargs)
+
+    def session(self):
+        return self
+
+    @staticmethod
+    def validate_port_name(port):
+        """Valid port names for this switch are of the form 1/0/1 or 1/2"""
+
+        val = re.compile(r'^\d+/\d+(/\d+)?$')
+        if not val.match(port):
+            raise BadArgumentError("Invalid port name. Valid port names for "
+                                   "this switch are of the from 1/0/1 or 1/2")
+        return
+
+    def disconnect(self):
+        pass
+
+    def modify_port(self, port, channel, new_network):
+        (port,) = filter(lambda p: p.label == port, self.ports)
+        interface = port.label
+
+        if channel == 'vlan/native':
+            if new_network is None:
+                self._remove_native_vlan(interface)
+            else:
+                self._set_native_vlan(interface, new_network)
+        else:
+            vlan_id = channel.replace('vlan/', '')
+            legal = get_network_allocator(). \
+                is_legal_channel_for(channel, vlan_id)
+            assert legal, "HIL passed an invalid channel to the switch!"
+
+            if new_network is None:
+                self._remove_vlan_from_trunk(interface, vlan_id)
+            else:
+                assert new_network == vlan_id
+                self._add_vlan_to_trunk(interface, vlan_id)
+
+    def revert_port(self, port):
+        self._remove_all_vlans_from_trunk(port)
+        if self._get_native_vlan(port) is not None:
+            self._remove_native_vlan(port)
+
+    def get_port_networks(self, ports):
+        response = {}
+        for port in ports:
+            response[port] = self._get_vlans(port.label)
+            native = self._get_native_vlan(port.label)
+            if native is not None:
+                response[port].append(native)
+
+        return response
+
+    def _get_vlans(self, interface):
+        """ Return the vlans of a trunk port.
+
+        Does not include the native vlan. Use _get_native_vlan.
+
+        Args:
+            interface: interface to return the vlans of
+
+        Returns: List containing the vlans of the form:
+        [('vlan/vlan1', vlan1), ('vlan/vlan2', vlan2)]
+        """
+
+        # It uses the REST API CLI which is slow but it is the only way
+        # because the switch is VLAN centric. Doing a GET on interface won't
+        # return the VLANs on it, we would have to do get on all vlans (if that
+        # worked reliably in the first place) and then find our interface there
+        # which is not feasible.
+
+        response = self._get_port_info(interface).splitlines()
+
+        try:
+            index = response.index("Vlanmembership:") + 3
```

Comment:
The 3 here is kind of troubling. Can we add some more checks here to make sure you're parsing out the right thing? You can also do a tighter regex to match exactly the line with the "T". Bounds checking on the array would be nice as well to avoid an out of bounds exception.

Reply:
Good call,  I should have been using regex all along.